scripts/find_heap_api_violations.py

```diff
@@ -94,6 +94,7 @@
         "std::shared_ptr<azure::storage_lite::shared_key_credential>",
         "std::shared_ptr<azure::storage_lite::storage_account>",
     ],
+    "thread_pool.h": ["std::shared_ptr<T>"],
```

Comment:
This should only be used if there is a reason why we can not use the tiledb-APIs, which is usually for compatibility with an external API. Why can't we use `tdb_shared_ptr`?

Reply:
There's no tdb-version for weak_ptr. I didn't want to audit the code to make sure everything's compatible. So, as a temporary measure, I just made an exception.